src/Routers/FilesRouter.js

```diff
@@ -69,35 +70,30 @@ export class FilesRouter {
   }
 
   createHandler(req, res, next) {
+    const config = req.config;
+    const filesController = config.filesController;
+    const filename = req.params.filename;
+    const contentType = req.get('Content-type');
+
     if (!req.body || !req.body.length) {
       next(
         new Parse.Error(Parse.Error.FILE_SAVE_ERROR, 'Invalid file upload.')
       );
       return;
     }
 
-    if (req.params.filename.length > 128) {
-      next(
-        new Parse.Error(Parse.Error.INVALID_FILE_NAME, 'Filename too long.')
-      );
-      return;
+    var error;
+    if (typeof filesController.adapter.validateFilename === 'function') {
+      error = filesController.adapter.validateFilename(filename);
+    } else {
+      error = validateFilename(filename);
```

Comment:
This is my cheat for an adapter which doesn't yet implement the method.   Not sure if TS compilation would of puked before this could happen.

Reply:
I don’t see an issue with this as we have this check for FileStreams.

Adapters like S3 currently does have this function.

Reply:
Instead I changed the FileController signature and then moved the check there.  Turns out I'm doing this specifically for the S3 Adapter, so I'm making sure it works with the old one.

Reply:
https://github.com/parse-community/parse-server-s3-adapter/pull/76